SonarRunner.Shim/SonarProjectPropertiesValidator.cs

```diff
@@ -0,0 +1,47 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SonarProjectPropertiesValidator.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using SonarQube.Common;
+
+namespace SonarRunner.Shim
+{
+    public static class SonarProjectPropertiesValidator
+    {
+        /// <summary>
+        /// Verifies that no sonar-project.properties conflicting with the generated one exists within the project
+        /// </summary>
+        /// <param name="sonarRunnerCwd">Solution folder to check</param>
+        /// <param name="projects">MSBuild projects to check, only valid ones will be verified</param>
+        /// <param name="onValid">Called when validation succeeded</param>
+        /// <param name="onInvalid">Called when validation fails, with the list of folders containing a sonar-project.properties file</param>
+        public static void Validate(string sonarRunnerCwd, IDictionary<ProjectInfo, ProjectInfoValidity> projects, Action onValid, Action<IList<string>> onInvalid)
+        {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Consider a design where 'SonarProjectPropertiesValidator.Validate(string, IDictionary<ProjectInfo, ProjectInfoValidity>, Action, Action<IList<string>>)' doesn't nest generic type 'Action<IList<string>>'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3ADoNotNestGenericTypesInMemberSignatures)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) In method 'SonarProjectPropertiesValidator.Validate(string, IDictionary<ProjectInfo, ProjectInfoValidity>, Action, Action<IList<string>>)', correct the spelling of 'Cwd' in parameter name 'sonarRunnerCwd' or remove it entirely if it represents any sort of Hungarian notation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AIdentifiersShouldBeSpelledCorrectly)